ansible-k8s-vsphere/tasks/main.yml

```diff
@@ -0,0 +1,426 @@
+- name: test vsphere
+  hosts: localhost
+  gather_facts: false
+  vars:
+    cluster_num_nodes: 6
+    cluster_num_control_plane: 3
+    cluster_name: CLUSTER-NAME
+    vm_cpu: 4
+    vm_mem: 15000
+    vm_timezone: America/Chicago
+    vm_disk_size: 128G
+    admin_ssh_user: admin
+    rke_ssh_user: rancher
+
+    govc_insecure: 1
+    govc_url: acmapp-vcenter.acm.cs
+    govc_username: USERNAME
+    govc_password: PASSWORD
+    govc_datastore: /ACMAPP/datastore/vsanDatastore
+    govc_network: ACMAPP-DSwitch-VM Network
+    govc_resource_pool: /ACMAPP/host/acmapp-cluster
+    govc_datacenter: /ACMAPP
+    govc_tag_category_cluster: "clusters"
+    govc_tag_category_role: "k8s_role"
+    vm_domain: acm.cs
+    base_work_dir: /home/bmiddha/.tmp
+    govc_version: 0.23.0
+    govc_download_url: https://github.com/vmware/govmomi/releases/download/v{{ govc_version }}/govc_linux_amd64.gz
+    govc_checksum: sha1:ba777988af807e4a9aaea47339125f15cdbc8eea
+    rke_version: 1.2.2
+    rke_download_url: "https://github.com/rancher/rke/releases/download/v{{ rke_version }}/rke_linux-amd64"
+    rke_checksum: "sha256:37b26104c0868706bc0c24e47a41c6dcee8e3ab6bc3b41425ade8c58337246f3"
+    ubuntu_version_name: focal
+    ubuntu_cloud_image: https://cloud-images.ubuntu.com/{{ ubuntu_version_name }}/current/{{ ubuntu_version_name }}-server-cloudimg-amd64.ova
+    ubuntu_cloud_image_checksum: sha256:197b7b126c0a16432d121c66022b0c7f8d591166150e7f9f311c2145b2b54a01 
+    template_name: "{{ cluster_name }}-template"
+    node_name: "{{ cluster_name }}-%d"
+    cluster_nodes_worker: []
+    cluster_nodes_control_plane: []
+    cluster_nodes: []
+
+  environment:
+    GOVC_INSECURE: "{{ govc_insecure }}"
+    GOVC_URL: "{{ govc_url }}"
+    GOVC_USERNAME: "{{ govc_username }}"
+    GOVC_PASSWORD: "{{ govc_password }}"
+    GOVC_DATASTORE: "{{ govc_datastore }}"
+    GOVC_NETWORK: "{{ govc_network }}"
+    GOVC_RESOURCE_POOL: "{{ govc_resource_pool }}"
+    GOVC_DATACENTER: "{{ govc_datacenter }}"
+    GOVC_FOLDER: "{{ cluster_name }}"
+
+  tasks:
+
+  - name: Build cluster nodes (cluster_nodes) array
+    set_fact: 
+      cluster_nodes: "{{ cluster_nodes | default([]) + [item] }}"
+    with_sequence: "start={{ 1 }} end={{ cluster_num_nodes }} stride=1 format={{ node_name }}"
+
+  - name: Build cluster control plane nodes (cluster_nodes_control_plane) array
+    set_fact: 
+      cluster_nodes_control_plane: "{{ cluster_nodes_control_plane | default([]) + [item] }}"
+    with_sequence: "start={{ 1 }} end={{ cluster_num_control_plane }} stride=1 format={{ node_name }}"
+
+  - block:
+    - name: Bandaid
```

Comment:
:adhesive_bandage: This is for your inventory edge-case, right? Where it complains about the last node, right?

Reply:
Yes. I think `with_sequence` is evaluated before `when`. It complains it since `start` > `end` when `cluster_num_control_plane == cluster_num_nodes `


```yml
    - name: Build cluster worker nodes (cluster_nodes_worker) array
      set_fact: 
        cluster_nodes_worker: "{{ cluster_nodes_worker | default([]) + [item] }}"
      with_sequence: "start={{ cluster_num_control_plane + 1}} end={{ cluster_num_nodes }} stride=1 format={{ node_name }}"
      when: "{{ cluster_num_nodes > cluster_num_control_plane }}"
```